src/include/homestore/index/index_table.hpp

```diff
@@ -250,7 +261,15 @@ class IndexTable : public IndexTableBase, public Btree< K, V > {
                           parent_node->node_id());
             return btree_status_t::not_found;
         }
-        BT_LOG(INFO, "Repairing node={} with last_parent_key={}", parent_node->to_string(),
+
+        // Get all original child ids as a support to check if we are beyond the last child node
+        std::set<bnodeid_t> orig_child_ids;
+        for (uint32_t i = 0; i < parent_node->total_entries(); ++i) {
+            BtreeLinkInfo link_info;
+            parent_node->get_nth_value(i, &link_info, true);
+            orig_child_ids.insert(link_info.bnode_id());
```

Comment:
what is the reason of having orig_child_ids?

Reply:
Refer to https://docs.google.com/presentation/d/1NUvu1ZlDKqJWfBayg2lSWHSB6TEaZ4Y3VbnGc996drM/edit#slide=id.g3068b64e06d_0_0

Corner case 2 & 3, since the first/last key can both change and exceed the original parent last key, we need this table to see if they should belong to the parent.